### PR TITLE
[Merged by Bors] - feat(source): added `trasnform` support for source connectors

### PR DESCRIPTION
**Pull request by galibey**
Added support of the `transform` section for all source connectors. The previous way of setting smartmodule for source connectors will be ignored.

Minor versions bump for source connectors. 
Patch versions bump for sink connectors since they didn't get any new functionality yet. 

Integration tests will be added after the initial review passes.

### CODE REVIEW

File: rust-connectors/sources/syslog/Cargo.toml [hunk 1]
```diff
@@ -8,7 +8,7 @@ edition = "2021"
 [dependencies]
 syslog_loose = "0.17.0"
 structopt = "0.3"
-fluvio = "0.14"
+fluvio = "0.15"
```
Comment (sehz) [inline rust-connectors/sources/syslog/Cargo.toml:11]: this can be changed to `0.1.0` since there was no release

**Review by sehz — CHANGES_REQUESTED** [review]
Looks good.  We don't need to bump up changes since haven't release official versions for bit.

File: rust-connectors/sinks/dynamodb/Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "dynamodb-sink"
-version = "0.3.1"
+version = "0.3.2"
```
[the remark this answers is not in the document]
Comment (galibey) [reply rust-connectors/sinks/dynamodb/Cargo.toml]: done

File: rust-connectors/sinks/kafka/Cargo.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "kafka-sink"
-version = "0.3.1"
+version = "0.3.2"
```
[the remark this answers is not in the document]
Comment (galibey) [reply rust-connectors/sinks/kafka/Cargo.toml]: done

File: rust-connectors/sinks/postgres/Cargo.toml [hunk 4]
```diff
@@ -1,13 +1,13 @@
 [package]
 name = "postgres-sink"
-version = "0.3.1"
+version = "0.3.2"
```
[the remark this answers is not in the document]
Comment (galibey) [reply rust-connectors/sinks/postgres/Cargo.toml]: done

File: rust-connectors/sinks/slack/Cargo.toml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "slack-sink"
-version = "0.3.1"
+version = "0.3.2"
```
[the remark this answers is not in the document]
Comment (galibey) [reply rust-connectors/sinks/slack/Cargo.toml]: done

File: rust-connectors/sinks/sql/Cargo.toml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "sql-sink"
-version = "0.1.0"
+version = "0.1.1"
```
[the remark this answers is not in the document]
Comment (galibey) [reply rust-connectors/sinks/sql/Cargo.toml]: done

File: rust-connectors/sources/http/Cargo.toml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "http-source"
-version = "0.4.0"
```
[the remark this answers is not in the document]
Comment (galibey) [reply rust-connectors/sources/http/Cargo.toml]: done

File: rust-connectors/sources/kafka/Cargo.toml [hunk 8]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "kafka-source"
-version = "0.3.1"
+version = "0.4.0"
 edition = "2021"
```
[the remark this answers is not in the document]
Comment (galibey) [reply rust-connectors/sources/kafka/Cargo.toml]: done

File: rust-connectors/sources/mqtt/Cargo.toml [hunk 9]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "mqtt-source"
-version = "0.5.0"
+version = "0.6.0"
```
[the remark this answers is not in the document]
Comment (galibey) [reply rust-connectors/sources/mqtt/Cargo.toml]: done

File: rust-connectors/sources/syslog/Cargo.toml (hunk 1 above)
Comment (galibey) [reply rust-connectors/sources/syslog/Cargo.toml:11]: do you mean the version of syslog connector? If so, better to do it separately.

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**